src/functions-and-arrays.js

```diff
@@ -106,7 +208,35 @@ const matrix = [
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
 
-function greatestProduct() {}
+function greatestProduct(matrix) {
+  let rowNumber = 0;
+  let columnNumber = 0;
+  let greatestResult = 0;
+  
+  for (let i = matrix [0][0]; i < 17; i++) {
```

Comment:
To iterate over all elements in the two-dimensional-array, you would need both loops nested inside of each other.